Algorithms/Search/InterpolationSearch.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Algorithms.Search;
+
+/// <summary>
+/// Interpolation search is an algorithm used for searching for a key in an array that has been ordered by numerical values.
+/// It's similar to a BinarySearch in the form of it truncating the array of which is searching though.
+/// There are a few things that the array needs to keep track of while running.
+/// L = The left most index of the array.
+/// R = The right most index of the array.
+/// Key = The value we're trying to find.
+/// It utilizes a formula which runs every iteration until it finds the key we're looking for.
+/// Formula: L + ((R-L) / (values[R] - values[L])) * (key - values[L]).
+/// </summary>
+public class InterpolationSearch
+{
+    /* Index of values to search from */
+    private readonly int[] indicies = { 1, 3, 5, 8, 10, 22, 31, 35, 37, 42, 51 };
+
+    /* The key (value) we want to find in the index of values */
+    private readonly int key = 22;
```

Comment:
These must be search parameters and not hard-coded

Reply:
Refactored.